backend/chromeApiServer/src/main/java/com/canvas/service/EvaluationService.java

```diff
@@ -0,0 +1,157 @@
+package com.canvas.service;
+
+import com.canvas.service.helperServices.CanvasClientService;
+import com.canvas.service.helperServices.FileService;
+import com.canvas.service.helperServices.ProcessExecutor;
+import com.canvas.service.models.CommandOutput;
+import com.canvas.service.models.ExtensionUser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class handles the Student side of the evaluation process.
+ */
+@Service
+public class EvaluationService {
+    private final FileService fileService;
+    private final CanvasClientService canvasClientService;
+    String MAKEFILE = "makefile";
+
+    /**
+     * Constructor for creating our instances of FileService and CanvasClientService
+     *
+     * @param fileService         File service instance
+     * @param canvasClientService Canvas client service instance
+     */
+    @Autowired
+    public EvaluationService(FileService fileService, CanvasClientService canvasClientService) {
+        this.fileService = fileService;
+        this.canvasClientService = canvasClientService;
+    }
+
+    /**
+     * This method will compile the student code. The files are extracted
+     * and the program is compiled.
+     *
+     * @param user  User object holding all components associated with a user
+     * @param files The files to compile
+     * @return      return a CommandOutput response
+     */
+    public ResponseEntity<CommandOutput> compileStudentCodeFile(ExtensionUser user, MultipartFile[] files) {
+
+
+        // Retrieve file json from Canvas
+        byte[] makefileBytes = new byte[0];
+        try {
+            makefileBytes = canvasClientService.fetchFileUnderCourseAssignmentFolder(user, MAKEFILE);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        // Write makefile to file
+        fileService.writeFileFromBytes(MAKEFILE, makefileBytes, user.getUserId());
+
+        // Write submitted code files
+        for (MultipartFile file : files) {
+            fileService.writeFileFromMultipart(file, user.getUserId());
+        }
+
+        // Compile the files and grab output
```

Comment:
this was refactored with the `compileCodeFiles` method, see [master](https://github.com/Canvas-Code-Capstone/Canvas-Code/blob/0c04d1b751abd6eeb81159d679b834582a1ed3c3/backend/chromeApiServer/src/main/java/com/canvas/chromeApiServer/ChromeApiController.java#L105)

Reply:
I'm a bit confused by this comment. In the master, line 155-173 are private helper methods. My refactor moved them to the EvaluationService since the code referencing them was originally in the api method.

Reply:
right I understand those methods were moved to `EvaluationService`. We can just replace line 67-74 and call `compileCodeFiles` instead, similar to the compile step in the instructor API (line 109)

Reply:
Ahh, I get it now. Updated that code to call the helper method.